ethereum/handler/ethereumHandler.go

```diff
@@ -84,32 +87,29 @@ func NewEthereumHandler(contracts map[string]*EthereumContract, srcNetworkId mod
 }
 
 // Connect creates a new connection to the Ethereum client
-func (eh *EthereumHandler) Connect(dialEndpoint string) error {
-	if eh.EthereumClient != nil {
-		eh.EthereumClient.Close()
-	}
-	if eh.EthereumRPC != nil {
-		eh.EthereumRPC.Close()
-	}
-	maxtries := 10
+func (eh *EthereumHandler) Connect(dialEndpoints []string) error {
 	var err error
-	for {
-		if maxtries == 0 {
-			return fmt.Errorf("could not connect to web3 endpoint")
-		}
-		eh.EthereumRPC, err = ethrpc.Dial(dialEndpoint)
-		if err != nil || eh.EthereumRPC == nil {
-			log.Warnf("cannot create an ethereum rpc connection: (%s), trying again ...", err)
-			time.Sleep(time.Second * 2)
-			maxtries--
-			continue
+	for _, endpoint := range dialEndpoints {
+		maxtries := 5
+		for {
+			if maxtries == 0 {
+				log.Warnf("could not connect to %s endpoint, trying the next one ...", endpoint)
```

Comment:
"one..." (or even better, remove the "..." since we are not using that in general)